src/pipeline.ts

```diff
@@ -487,30 +513,41 @@ export class GitHubWorkflow extends PipelineBase {
     };
   }
 
-  private stepsToConfigureAws({ region, assumeRoleArn }: { region: string; assumeRoleArn?: string }): github.JobStep[] {
-    const params: Record<string, any> = {
-      'aws-access-key-id': `\${{ secrets.${this.awsCredentials.accessKeyId} }}`,
-      'aws-secret-access-key': `\${{ secrets.${this.awsCredentials.secretAccessKey} }}`,
-      'aws-region': region,
-      'role-skip-session-tagging': true,
-      'role-duration-seconds': 30 * 60,
-    };
-
-    if (this.awsCredentials.sessionToken) {
-      params['aws-session-token'] = `\${{ secrets.${this.awsCredentials.sessionToken} }}`;
+  private stepsToConfigureAws(openId: boolean, { region, assumeRoleArn }: { region: string; assumeRoleArn?: string }): github.JobStep[] {
+    function getDeployRole(arn: string) {
+      return arn.replace('cfn-exec', 'deploy');
     }
 
-    if (assumeRoleArn) {
-      params['role-to-assume'] = assumeRoleArn;
-      params['role-external-id'] = 'Pipeline';
+    let steps: github.JobStep[] = [];
+
+    if (openId) {
+      steps.push(awsCredentialStep('Authenticate Via OIDC Role', {
+        region,
+        oidcRoleArn: this.awsOidcRoleArn,
+      }));
+
+      if (assumeRoleArn) {
+        // Result of initial credentials with OIDC role are these environment variables
+        steps.push(awsCredentialStep('Assume CDK Deploy Role', {
+          region,
+          accessKeyId: '${{ env.AWS_ACCESS_KEY_ID }}',
+          secretAccessKey: '${{ env.AWS_SECRET_ACCESS_KEY }}',
+          sessionToken: '${{ env.AWS_SESSION_TOKEN }}',
+          roleToAssume: getDeployRole(assumeRoleArn),
+        }));
+      }
+    } else {
+      steps.push(awsCredentialStep('Authenticate Via GitHub Secrets', {
+        region,
+        accessKeyId: `\${{ secrets.${this.awsCredentials.accessKeyId} }}`,
+        secretAccessKey: `\${{ secrets.${this.awsCredentials.secretAccessKey} }}`,
+        sessionToken: `\${{ secrets.${this.awsCredentials.sessionToken} }}`,
+        roleToAssume: assumeRoleArn,
+        roleExternalId: assumeRoleArn ? 'Pipeline' : undefined,
```

Comment:
Why a hardcoded externalId here btw?

Reply:
talked about not using externalID anywhere because it does not add security for our use case